### ISSUE TITLE
Bug: Woo Subscriptions renewal payments restart memberships on PMPro 3.0

### ISSUE THREAD

**Issue by dwanjuki** [strangerstudios/pmpro-woocommerce#201]
**Describe the bug**
When a subscription renewal payment is made, the members current membership is ended, and they are given the level anew.

A side effect of this is that PMPro series progress is lost.

The issue does not happen on PMPro 2.12.10.

**To Reproduce**
1. Steps to reproduce the behavior:
2. Create a subscription product that grants a membership level then buy it.
3. Navigate to _WooCommerce > Subscriptions_ and note the subscription ID.
4. Navigate to _WooCommerce > Status > Scheduled Actions_, search for the subscription ID, and run the scheduled subscription payment task.
5. Check Membership Level History.
6. See that a new entry is created in level history with current day as start date.

**Screenshots**
![Capture](https://github.com/strangerstudios/pmpro-woocommerce/assets/123358258/99b184ea-08a8-4b65-8acc-193b72812662)

**Expected behavior**
There shouldn't be a level change when Woo Subs renewals are processed.

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only Paid Memberships Pro plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Memberlite](https://wordpress.org/themes/memberlite/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
WP 6.5
PMPro 3.0.1
WooCommerce 8.7.0
PMPro WooCommerce 1.9
Woo Subscriptions 6.1.0
WooCommerce Stripe Gateway 8.1.0
</details>

**Comment by andrewlimaza** [comment]
I made more progress on this, I managed to get a different status from "active" to "changed" but I can confirm the start date stayed the same.

To test this:
1. Purchase the first level, adjust the levels' start date for this member to an older date (i.e. 1 January 2020)
2. Run the subscriptions action scheduler and see that the start date remains 1 January 2020 (which shouldn't reset things).

This would unnecessarily add rows to the database and we should resolve this for cases with Woo Subscriptions but I think this is due to the subscription triggering an order which then triggers a new order and status changes which then adjusts the membership data as per our tests.

If the start date is changing to the date of the most recent order means this is broken and not as intended based on looking into the code for reference.
